### PR TITLE
edge-23.4.3

### PR DESCRIPTION
**Pull request by adleong**
This edge release improves compatibility with ArgoCD by changing the Linkerd
control plane to create Lease resources at runtime rather than including them
in the Helm chart. It also addresses a CVE by upgrading an underlying
dependency.

* Upgraded `h2` dependency to address CVE-2023-26964
* Fixed an issue where `server_port_subscribers` metric in the Destination
  controller was sometimes absent
* Removed the policy-controller-write Lease from the control plane Helm chart in
  favor of creating it at runtime
* Updated the proxy-injector to pass opaque port lists to the proxy as ranges
  rather than individually, greatly reducing the size of proxy manifests when
  large opaque port ranges are set
* Fixed an issue where the proxy was performing protocol detection on ports
  marked as opaque
* Improved backwards compatibility between 2.13 proxies and 2.12 control planes

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,5 +1,21 @@
...
+
+* Upgraded `h2` dependency to address CVE-2023-26964
+* Fixed an issue where `server_port_subscribers` metric in the Destination
+  controller was sometimes absent
+* Removed the policy-controller-write Lease from the control plane Helm chart in
+  favor of creating it runtime
+* Updated the proxy-injector to pass opaque port lists to the proxy as ranges
+  rather than individually, greatly reducing the size of proxy manifests when
+  large opaque port ranges are set
+
```
Comment (hawkw) [inline CHANGES.md:18]: I think some of the proxy changes from [v2.200.0](https://github.com/linkerd/linkerd2/pull/10822) are probably worth listing here. In particular:
- https://github.com/linkerd/linkerd2-proxy/pull/2397, which fixes #10745 
- https://github.com/linkerd/linkerd2-proxy/pull/2396, which is part of the 2.12 back-compatibility story
- and possibly https://github.com/linkerd/linkerd2-proxy/pull/2395, although the last bullet point about the proxy injector change kind of addresses this...

File: CHANGES.md [hunk 2]
```diff
@@ -1,5 +1,24 @@
...
+This edge release improves compatibility with ArgoCD by changing the Linkerd
+control plane to create Lease resources at runtime rather than including them
+in the Helm chart. It also addresses a CVE by upgrading an underlying
+dependency.
+
+* Upgraded `h2` dependency to address CVE-2023-26964
+* Fixed an issue where `server_port_subscribers` metric in the Destination
+  controller was sometimes absent
+* Removed the policy-controller-write Lease from the control plane Helm chart in
+  favor of creating it runtime
```
Comment (alpeb) [inline CHANGES.md:14]:
```suggestion
  favor of creating it at runtime
```

**Review by alpeb — COMMENTED** [review]
With `edge-23.4.2` we bumped the default cni plugin version to `v1.1.1` and I forgot to bump the linkerd2-cni chart version. Can you do that here? I think the new version should be `30.8.0-edge`.

**Review by hawkw — APPROVED** [review]

**Review by alpeb — APPROVED** [review]
LGTM :+1:

**Pull request merged**